ray-operator/controllers/ray/rayservice_controller.go

```diff
@@ -98,65 +104,112 @@ func (r *RayServiceReconciler) Reconcile(ctx context.Context, request ctrl.Reque
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	var rayClusterInstance *rayv1alpha1.RayCluster
-	if rayClusterInstance, err = r.getOrCreateRayClusterInstance(ctx, rayServiceInstance); err != nil {
+	var runningRayClusterInstance *rayv1alpha1.RayCluster
+	var preparingRayClusterInstance *rayv1alpha1.RayCluster
+	if runningRayClusterInstance, preparingRayClusterInstance, err = r.reconcileRayCluster(ctx, rayServiceInstance); err != nil {
 		err = r.updateState(ctx, rayServiceInstance, rayv1alpha1.FailToGetOrCreateRayCluster, err)
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	log.Info("Updated RayCluster")
+	rayServiceLog.Info("Done reconcileRayCluster")
+	if errStatus := r.Status().Update(ctx, rayServiceInstance); errStatus != nil {
+		rayServiceLog.Error(errStatus, "Fail to update status of RayService", "rayServiceInstance", rayServiceInstance)
+		return ctrl.Result{}, err
+	}
+
+	if rayServiceInstance.Status.PreparingRayClusterName != "" && preparingRayClusterInstance == nil {
```

Comment:
What's PreparingRayClusterName, again?

Reply:
OH! "The name of the Ray Cluster which is being prepared."

Reply:
yes, I should use a better name

Reply:
Maybe `PendingRayClusterName`.

On first read, I parsed the meaning of `Status.PreparingRayClusterName` as "The controller is currently preparing a Ray cluster's name."